tethys_gizmos/templatetags/tethys_gizmos.py

```diff
@@ -77,6 +77,39 @@ def default(self, obj):
         return super().default(obj)
 
 
+class SetVarNode(template.Node):
+
+    def __init__(self, var_name, var_value):
+        self.var_names = var_name.split('.')
+        self.var_name = self.var_names.pop()
+        self.var_value = var_value
+
+    def render(self, context):
+        try:
+            value = template.Variable(self.var_value).resolve(context)
+        except template.VariableDoesNotExist:
+            value = ''
+
+        for name in self.var_names:
+            context = context[name]
+
+        context[self.var_name] = value
+
+        return ''
+
+
+@register.tag(name='set')
```

Comment:
Is there a reason you don't want to use the `with` tag to set variables in the template?